src/olympia/users/admin.py

```diff
@@ -287,18 +287,28 @@ def has_active_api_key(self, obj):
         return obj.api_keys.filter(is_active=True).exists()
     has_active_api_key.boolean = True
 
-    def collections_created(self, obj):
+    def collections_authorship(self, obj):
         return related_content_link(obj, Collection, 'author')
-    collections_created.short_description = _('Collections')
-
-    def addons_created(self, obj):
-        return related_content_link(obj, Addon, 'authors',
-                                    related_manager='unfiltered')
-    addons_created.short_description = _('Addons')
+    collections_authorship.short_description = _('Collections')
+
+    def addons_authorship(self, obj):
+        non_deleted_authorship_count = Addon.unfiltered.filter(
+            addonuser__user=obj).exclude(
+            addonuser__role=amo.AUTHOR_ROLE_DELETED).count()
+        deleted_authorship_count = Addon.unfiltered.filter(
+            addonuser__user=obj,
+            addonuser__role=amo.AUTHOR_ROLE_DELETED).count()
```

Comment:
I don't suppose there's a way to do this in a single query?  Something like `.values_list('addonuser__role', flat=True)` and count the roles in the response?

Reply:
I'd like to avoid doing this in case there are a lot of results. But there might be a clean way with some aggregation trickery, I'm not sure (it might not work in MySQL). I'll check.

Reply:
`.aggregate(Count('role'))` maybe?

Reply:
ae19826171ff10f1aed282c53f351b75e58b26d4